src/providers/validationProvider.ts

```diff
@@ -32,8 +33,22 @@ export async function doValidate(
       validationManager.getValidationFromCache(textDocument.uri) ||
       new Map<string, Diagnostic[]>();
   } else {
-    // full validation with ansible-lint
-    diagnosticsByFile = await context.ansibleLint.doValidate(textDocument);
+    // full validation with ansible-lint or ansible syntax-check (if ansible-lint is not installed or disabled)
+
+    const lintAvailability = getExecutablePath('ansible-lint');
```

Comment:
Why are you trying to await in separate places multiple times? Just await once here and reuse the result in other places. What you did is quite untypical.
```suggestion
    const lintAvailability = await getExecutablePath('ansible-lint');
```

Reply:
Also, it's probably reasonable to expect this to be a string passed from a VS Code config rather than just a hardcoded value.

Reply:
Supporting config variable is out of the scope as of now. It can be viewed as a separate task and I am going to look into it.

Reply:
Created an issue to track changes. https://github.com/ansible/ansible-language-server/issues/14

Reply:
Supporting config variable is not optional - unless it is implemented it will cause a regression and disable linting for any users that do use non-default value.

Reply:
And that variable is already there, so it's just a matter of using it.

Reply:
@tomaciazek mind sending a suggested change? Otherwise, I'm okay with adding this in a follow-up PR.

Reply:
So you're "okay" with knowingly breaking things (for no good reason) until a follow-up PR? Master is supposed to be "always deployable", or at least that's what we should strive for, so no, this is not "okay".

And here's all it takes to fix it:
```typescript
const settings = await context.documentSettings.get(textDocument.uri);
const lintAvailability = await getExecutablePath(settings.ansibleLint.path);
```

Reply:
@tomaciazek, thanks for your suggestion. This change has been implemented in the latest commit.